pandas/core/generic.py

```diff
@@ -2399,6 +2413,13 @@ def to_json(
             date_format = "iso"
         elif date_format is None:
             date_format = "epoch"
+
+        config.is_nonnegative_int(indent)
+        if indent is None:
+            int_indent = 0
+        else:
+            int_indent = indent
```

Comment:
Can't we collapse this to:

~~~python
indent = indent or 0
~~~

BTW, at this point, we don't need another variable because the `config` call already confirms an `int` or `None`.

Reply:
It's valid Python but doesn't jive with typing, because `indent` local to this function is `Optional[int]` but everywhere else it is just `int`. This is an easy way to limit the types for the checker, since `int_indent` will always be `int`

Reply:
There is a flag that supports this:

https://github.com/python/mypy/pull/6197

But not enabled, assuming we even wanted that

Reply:
Actually I stand corrected - looks like it can narrow the type here implicitly. Nice catch!